sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java

```diff
@@ -427,6 +438,67 @@ private CarbonFileInputFormat prepareFileInputFormat(Job job, boolean enableBloc
     }
   }
 
+  private <T> void totalRowCountInSplits(Job job, List<InputSplit> splits,
+      List<Long> rowCountInSplit)
+      throws IOException, InterruptedException {
+    CarbonFileInputFormat format = this.prepareFileInputFormat(job, false, true);
+    long sum = 0;
+    boolean isIUDTable = false;
+    if (!StringUtils.isEmpty(this.tablePath)) {
+      // Get store path of the table.
+      String schemaFilePath = CarbonTablePath.getSchemaFilePath(this.tablePath, this.hadoopConf);
+      // If metadata folder exists, it is a transactional table
+      CarbonFile schemaFile = FileFactory.getCarbonFile(schemaFilePath, this.hadoopConf);
+      boolean isTransactionalTable = schemaFile.exists();
+      // Check if update or delete happened on the table.
+      if (!isTransactionalTable) {
+        CarbonFile emptyMetadataFile = FileFactory.getCarbonFile(this.tablePath +
```

Comment:
same comment as above, I think this method is called for non-transactional table only

Reply:
done